packages/material-ui-utils/src/refType.js

```diff
@@ -0,0 +1,10 @@
+import PropTypes from 'prop-types';
+
+const refType = PropTypes.oneOfType([
+  PropTypes.func,
+  PropTypes.shape({
```

Comment:
This would create a redundant warning if the shape doesn't match. React itself already warns if the shape is different.  We don't need to repeat this with prop-types.

Reply:
It's good to know. Is they any potential for using `Proptypes.any`? I mean, I believe we always rely on React.useImperativeHandle or <native ref>, can *react* do the job instead of *prop-types*?

Reply:
It could but doesn't currently. It only considers functions and objects (+shape checking). The rest is ignored.

Reply:
Ok thanks. So we shouldn't use the prop types, we should keep using what we had before.